packages/fast-components-react-base/src/stack-panel/stack-panel.tsx

```diff
@@ -0,0 +1,738 @@
+}
+
+class StackPanel extends Foundation<
+    StackPanelHandledProps,
+    StackPanelUnhandledProps,
+    StackPanelState
+> {
+    public static displayName: string = `${DisplayNamePrefix}StackPanel`;
+
+    public static defaultProps: Partial<StackPanelProps> = {
+        managedClasses: {},
+        enableVirtualization: true,
+        neverVirtualizeIndexes: [],
+        defaultItemSpan: 100,
+        preloadBufferLength: 1,
+        orientation: stackPanelOrientation.vertical,
+        nextItemPeek: 50,
+        itemSpanOverrides: {},
+        enableSmoothScrolling: true,
+        scrollDuration: 500,
+    };
+
+    private static DirectionAttributeName: string = "dir";
+
+    protected handledProps: HandledProps<StackPanelHandledProps> = {
+        nextItemPeek: void 0,
+        defaultItemSpan: void 0,
+        enableVirtualization: void 0,
+        neverVirtualizeIndexes: void 0,
+        itemSpanOverrides: void 0,
+        preloadBufferLength: void 0,
+        orientation: void 0,
+        enableSmoothScrolling: void 0,
+        scrollDuration: void 0,
+        managedClasses: void 0,
+        initiallyVisibleItemIndex: void 0,
+    };
+
+    private rootElement: React.RefObject<HTMLDivElement> = React.createRef<
+        HTMLDivElement
+    >();
+
+    private itemContainerElement: React.RefObject<HTMLDivElement> = React.createRef<
+        HTMLDivElement
+    >();
+
+    private openRequestAnimationFrame: number = null;
+    private resizeDetector: ResizeObserverClassDefinition;
+    private itemPositions: ItemPosition[];
+    private viewPortSpan: number = 0;
+    private itemContainerSpan: number = 0;
+
+    /**
+     *  The position the current scroll animation started at
+     */
+    private currentScrollAnimStartPosition: number;
+
+    /**
+     * The target position of the current scroll animation
+     */
+    private currentScrollAnimEndPosition: number;
+
+    /**
+     * Start time for current scroll animation
+     */
+    private currentScrollAnimStartTime: number;
+
+    /**
+     * Flag indicates if a scroll animation is in progress
+     */
+    private isScrollAnimating: boolean = false;
+
+    /**
+     * Stores last scroll position from scroll events
+     */
+    private lastRecordedScroll: number = 0;
+
+    /**
+     * constructor
+     */
+    constructor(props: StackPanelProps) {
+        super(props);
+
+        this.updateItemPositions();
+
+        this.state = {
+            disabled: false,
+            renderedRangeStartIndex: 0,
+            renderedRangeEndIndex: 0,
+            direction: Direction.ltr,
+            isScrolling: false,
+        };
+    }
+
+    /**
+     * Renders the component
+     */
+    public render(): React.ReactElement<HTMLDivElement> {
+        const {
+            stackPanel_itemContainer,
+        }: StackPanelClassNameContract = this.props.managedClasses;
+
+        return (
+            <div
+                {...this.unhandledProps()}
+                tabIndex={-1}
```

Comment:
Why do both of these elements need tabIndex?

Reply:
removed, may not be necessary